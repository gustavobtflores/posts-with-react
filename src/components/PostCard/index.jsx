import styled from 'styled-components';

const PostsWrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 16px;
`;

const Post = styled.div`
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  padding: 16px;
  transition: transform 0.2s ease-in-out;

  &:hover {
    transform: scale(1.02);
  }
`;

const PostTitle = styled.h2`
  font-size: 2rem;
  font-weight: 600;
  margin-bottom: 1rem;
`;

const PostBody = styled.p``;

const PostCover = styled.img`
  max-width: 100%;
`;

const PostCard = ({ posts }) => {
  <PostsWrapper>
    {posts.map((post) => (
      <Post key={post.id}>
        <PostCover src={post.cover} alt={post.title} />
        <PostTitle>{post.title}</PostTitle>
        <PostBody>{post.body}</PostBody>
      </Post>
    ))}
  </PostsWrapper>;
};

export default PostCard;
