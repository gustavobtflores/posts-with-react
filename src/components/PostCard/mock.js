const postCardMock = [
  {
    id: 1,
    title: 'title1',
    cover: 'img/img.png',
    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  },
];

export default postCardMock;
