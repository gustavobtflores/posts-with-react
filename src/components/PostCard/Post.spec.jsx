import { render, screen } from '@testing-library/react';
import PostCard from '.';
import mock from './mock';

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard posts={mock} />);

    expect(screen.getByRole('img', { name: mock.title })).toHaveAttribute('src', mock.img);

    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();

    expect(screen.getByText('Lorem ipsum dolor sit amet, consectetur adipiscing elit.')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard posts={mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
