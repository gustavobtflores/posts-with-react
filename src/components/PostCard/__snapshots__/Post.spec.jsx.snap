// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ ixKPMO"
>
  <div
    class="sc-gsnTZi eQKhJm"
  >
    <img
      alt="title1"
      class="sc-eCYdqJ gNiPgP"
      src="img/img.png"
    />
    <h2
      class="sc-dkzDqf lhODJe"
    >
      title1
    </h2>
    <p
      class="sc-hKMtZM"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
  </div>
</div>
`;
